### PR TITLE
Popover Fix

### LINKED ISSUE #74: cells.Popover doesn't let me select text on the Popover window
I am using the Popover to display a cell that may potentially contain a lot of text in an abbreviated way in a Table. When I click on the cell with the abbreviated info, the Popover overlays a window with the full contents of the cell. This works as expected, so far. 

However, when I try to select text from the overlay window of the Popover (in order to copy it and work on it somewhere else), the overlay window disappears before I am able to grab the text. It would be good to not remove the popover window when we click on top of it.

### PR DESCRIPTION
**Pull request by darth-cheney**
This PR implements the behavior desired for Popovers in issue #74  

## Motivation and Context

Popover content areas would disappear if you clicked in them. For, this meant you could not highlight text inside Popover content when it was active since it would vanish on click. 

## Approach

Clicking elsewhere on the document or on the original Popover toggler will now cause the Popover to vanish. Clicking in the Popover content area, when it is shown, does nothing.
  
### Popover Component ###
With this PR, the Popover component adds/removes event handlers as needed depending on the state of the popover. This allows us to implement the desired behavior. For more detail, see the custom methods in the Popover.js file.
  
### NewCellHandler Updates ###
Until now we had been relying on a method called `updatePopovers()` inside of the NewCellHandler. This functionality has been removed, and is now the sole responsibility of any single Popover component.

## How Has This Been Tested?

Tested in the UI. We have added a CellsDemo example for Popover and also four tests for the demo.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

  
## Closes
#74

### CODE REVIEW

**Comment by atzannes** [comment]
After a `make clean; make install` I got the right behavior on Firefox 72.0.2 (64-bit) and on Chrome Version 79.0.3945.130 (Official Build) (64-bit).

However, with Chrome I had a very strange behavior where it would do the wrong thing (close the Popover when clicking on it) if I opened a new tab to test the behavior, whereas it would do the right thing if I opened it in a new window. Very odd. Anyway, that sounds like there could be something wrong with my setup. 

So from my end this looks like it solves the issue :+1:

**Comment by darth-cheney** [comment]
> However, with Chrome I had a very strange behavior where it would do the wrong thing (close the Popover when clicking on it) if I opened a new tab to test the behavior, whereas it would do the right thing if I opened it in a new window. Very odd.
  
This could just be some browser caching that happened with the JS resources. Or it could be something else still. Let's go ahead and roll with the PR, if you approve, and we can circle back to it if it continues to be an issue.

**Review by atzannes — APPROVED** [review]
I don't have enough background and context to pass judgement on the implementation, but the popover seems to behave as expected, and we have unit-tests for it, so :+1: from me!

File: object_database/web/cells_demo/popover.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#   Copyright 2017-2019 object_database Authors
```
Comment (atzannes) [inline object_database/web/cells_demo/popover.py:1]: I think this should be Copyright 2020, but who cares right?

**Pull request merged**